### ISSUE TITLE
Add support for testing coverage with custom stats

### ISSUE THREAD

**Issue by jnields** [ai/browserslist#219]
It is very useful to test a query's coverage percentage against caniuse data.

If a user is using custom stats, it may be even more beneficial to check the coverage against custom stats instead of the caniuse stats.

E.g.,

```
const browsers = browserslist('>.5% in my stats', { stats });
const coverageInUs = browserslist.coverage(browsers, 'US');
const coverageOfUsers = browserslist.coverage(browsers, 'my stats');
```

**Comment by ai** [comment]
For sure! I will try to add `browserslist.coverage(browsers, { stats })` support on this weekend.

**Comment by ai** [comment]
Done 4748478

**Comment by ai** [comment]
Released in 3.0.
